app/src/main/java/io/ipoli/android/shop/fragments/CoinsStoreFragment.java

```
@@ -0,0 +1,253 @@
+
+        ((MainActivity) getActivity()).initToolbar(toolbar, R.string.title_fragment_store);
+
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
+        layoutManager.setOrientation(LinearLayoutManager.VERTICAL);
+        coinsList.setLayoutManager(layoutManager);
+        coinsList.setEmptyView(rootLayout, R.string.empty_inbox_text, R.drawable.ic_inbox_grey_24dp);
+        adapter = new CoinsStoreAdapter(new ArrayList<>(), eventBus);
+        coinsList.setAdapter(adapter);
+
+        if (!NetworkConnectivityUtils.isConnectedToInternet(getContext())) {
+            showFailureMessage(R.string.no_internet_to_buy_coins);
+        } else {
+            iabHelper = new IabHelper(getContext(), BillingConstants.getAppPublicKey());
+            iabHelper.startSetup(result -> {
+                if (!result.isSuccess() || iabHelper == null) {
+                    showFailureMessage(R.string.something_went_wrong);
+                    return;
+                }
+                queryInventory();
+            });
+        }
+
+        return view;
+    }
+
+    private void showFailureMessage(int messageRes) {
+        progressBar.setVisibility(View.GONE);
+        failureMessage.setText(messageRes);
+        failureMessage.setVisibility(View.VISIBLE);
+    }
+
+    private void queryInventory() {
+        try {
+            ArrayList<String> skuList = new ArrayList<>();
+            skuList.add(SKU_COINS_100);
+            skuList.add(SKU_COINS_300);
+            skuList.add(SKU_COINS_500);
+            skuList.add(SKU_COINS_1000);
+            iabHelper.queryInventoryAsync(true, skuList, (result, inv) -> {
+                if (!result.isSuccess() || iabHelper == null) {
+                    showFailureMessage(R.string.something_went_wrong);
+                    return;
+                }
+                initItems(inv);
+            });
+        } catch (IabHelper.IabAsyncInProgressException e) {
+            e.printStackTrace();
```

We can report this error with Firebase crash reporting as well